lua/cspell/helpers.lua

```diff
@@ -58,18 +35,95 @@ M.create_cspell_json = function(params)
         path = cspell_json_file_path,
     }
 
-    CONFIG_INFO_BY_CWD[params.cwd] = info
-
+    CONFIG_INFO_BY_PATH[cspell_json_file_path] = info
     return info
 end
 
+local set_create = function(itable)
+    local set = {}
+    for _, value in pairs(itable) do
+        set[value] = true
+    end
+    return set
+end
+
+local set_compare = function(expected_values, new_values)
+    for key, _ in pairs(expected_values) do
+        if new_values[key] == nil then
+            return false
+        end
+    end
+    return true
+end
+
+--- create a merged cspell.json file that imports all cspell configs defined in cspell_config_dirs
+---@param params GeneratorParams
+---@param cspell_config_mapping table<number|string, string>
+---@return CSpellConfigInfo
+M.create_merged_cspell_json = function(params, cspell_config_mapping)
+    local vim_cache = vim.fn.stdpath("cache")
+    local plugin_name = "cspell.nvim"
+    local merged_config_key = vim.fn.fnamemodify(params.cwd, ":p:h:t")
```

Comment:
This cache key should be a bit more robust, if the user opens two or more directories with the same name, the cache will be clobbered by the latest directory.

We should URL encode the full cwd as the key.

Reply:
I am slightly hesitant on this change in case it causes users to have issues with filepath length(i.e windows). The suggestion makes sense and I agree that using just the name of the directory that neovim was opened in is probably not sufficient. 

I was trying to consider alternatives such as generating a fixed length hash of the directory path(adds complexity, and makes debugging difficult), or encoding the full path(see the above concern on filepath limits), but neither seemed like a great solution. I have gone ahead and implemented it with a full path encoding, but am open to hear any other suggestions that have less risk/concern around path limitations.

Reply:
I wouldn't be too worried about that limit, Vim already does something like that, it's not URL encoding, but it does keep full paths as keys for the swap files, replacing directory delimiters with a `%`.

See: `:h swap-file`

You can also see a list of your current swap files in: `~/.local/state/nvim/swap`.

Reply:
Sounds good, I guess if its already being done in the editor then its not really a breaking behavior/change. I updated the delimiter from `\\` to `%` to be in line with vims filepath encoding usage.